Add ke17z basic support

ankeXiao opened the pull request.

hakehuang commented:
@ankeXiao , there are several build failure

<html>
<body>

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
mgmt.mcumgr.fs.mgmt.hash.supported.crc32 | v3.6.0-3542-g8c0bff829db0 | Build failure | frdm_ke17z | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.sha256 | v3.6.0-3542-g8c0bff829db0 | Build failure | frdm_ke17z | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
mgmt.mcumgr.fs.mgmt.hash.supported.all | v3.6.0-3542-g8c0bff829db0 | Build failure | frdm_ke17z | tests/subsys/mgmt/mcumgr/fs_mgmt_hash_supported
sample.drivers.flash.shell | v3.6.0-3542-g8c0bff829db0 | Build failure | frdm_ke17z | samples/drivers/flash_shell


</body>
</html>
`

hakehuang commented:
now build and board testing are clean.

ankeXiao commented:
The soc module files depends on this PR "[pull/71670](https://github.com/zephyrproject-rtos/zephyr/pull/71670)" and is shared with KE15Z platform, so will be updated after it is reviewed and merged.

dleach02 commented on boards/nxp/frdm_ke17z/doc/frdmke17z.png:
as commented on your other PR, convert to webp and put through [tinypng.com](https://tinypng.com/)

dleach02 commented:
Please address the compliance error

ankeXiao committed:
soc: nxp: kinetis: add soc support for mke17z7

Added soc support for mke17z7

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

ankeXiao committed:
dts: arm: nxp: add ke17z and ke17z512 dtsi file

Included the nxp_ke1xz.dtsi file to use the same peripheral info,
and overwrite it for different peripherals

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

ankeXiao committed:
boards: nxp: frdm_ke17z: add the board support

Added configuration file to support frdm_ke17z board

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

ankeXiao committed:
drivers: clock_control: add a soc config macro wrap flexbus clock

ke17z7 platform does not define flexbus clock-related macros,
so comment it out.

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

ankeXiao committed:
tests: drivers: gpio: test gpio_basic_api for frdm_ke17z

Added frdm_ke17z.overlay to support testing gpio_basic_api sample

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

ankeXiao commented:
Hi all, there is a compilation failure in the "build-docs" under the "Documentation Build(HTML)" step, I have no idea which file  caused it, please help to check. Thanks!

dleach02 commented:
There was a site-wide CI failure going on. Appears to have been addressed.

danieldegrasse commented on boards/nxp/frdm_ke17z/frdm_ke17z_defconfig:
```diff
@@ -0,0 +1,20 @@
+
+# Clock Control
+CONFIG_CLOCK_CONTROL=y
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=72000000
```
Is this accurate? The documentation states the board is clocked at 48MHz, and this value should be equal to the clock frequency the SYSTICK is running at on boards using it (which this M0+ will by default). A good way to check if this value is correct is to use a sample like `samples/basic/blinky`, and make sure the LED toggles once per second using an oscilloscope or signal analyzer

ankeXiao replied:
Thanks for reminding, the actual frequency should be 48000000, which should be consistent with the index.rsy document.

ankeXiao commented on boards/nxp/frdm_ke17z/doc/index.rst:
```diff
@@ -0,0 +1,198 @@
+System Clock
+============
+
+The KE17Z SoC is configured to run at 48 MHz using the FIRC.
```
Corrected the value of SYS_CLOCK_HW_CYCLES_PER_SEC to 48MHZ.

danieldegrasse commented on dts/arm/nxp/nxp_ke17z.dtsi:
How much of this file duplicates the `ke1xz.dtsi` file? From what I can tell (just glancing, I may be missing differences), we are duplicating the following nodes:
- cpu
- pinctrl
- pcc
- lpuart0-2
- porta-e
- gpioa-e

So we could import the `ke1xz.dtsi` file here, and override/define the following nodes:
- sram_l
- sram_u
- scg
- ftfa


Does this seem right?

ankeXiao replied:
Only sram and ftfe currently, the scg can be consistent.

dleach02 commented on dts/arm/nxp/nxp_ke17z512.dtsi:
```diff
@@ -0,0 +1,47 @@
+				write-block-size = <8>;
+			};
+		};
+    };
```
tab instead of space

ankeXiao replied:
Thanks!
